test/stdio/lines.js

```diff
@@ -0,0 +1,94 @@
+import {scheduler} from 'node:timers/promises';
+import test from 'ava';
+import {execa} from '../../index.js';
+import {setFixtureDir} from '../helpers/fixtures-dir.js';
+import {getStdio} from '../helpers/stdio.js';
+
+setFixtureDir();
+
+const bigLine = '.'.repeat(1e5);
+const manyChunks = Array.from({length: 1e5}).fill('.');
+
+const inputGenerator = async function * (input, chunks) {
+	// eslint-disable-next-line no-unused-vars
+	for await (const chunk of chunks) {
+		for (const inputItem of input) {
+			yield inputItem;
+			// eslint-disable-next-line no-await-in-loop
+			await scheduler.yield();
```

Comment:
This line ensures that each `yield` emits its own chunk of data. Node.js would otherwise merge those into a single one, for performance reasons.